### PR TITLE
Update chainsaw in makefile

### PR DESCRIPTION
**Pull request by pavolloffay**
**Description:** <Describe what has changed.>

**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #issue-number

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -466,7 +466,7 @@ KUSTOMIZE_VERSION ?= v5.0.3
 CONTROLLER_TOOLS_VERSION ?= v0.16.1
 GOLANGCI_LINT_VERSION ?= v1.57.2
 KIND_VERSION ?= v0.20.0
-CHAINSAW_VERSION ?= v0.2.5
+CHAINSAW_VERSION ?= v0.2.11
```
Comment (pavolloffay) [inline Makefile:469]: it aligns with version used in GHA

**Review by swiatekm — APPROVED** [review]
Ideally, we should get renovate to bump versions in the Makefile for us. We're pretty bad at keeping it up ourselves.

**Review by jaronoff97 — APPROVED** [review]

**Review by frzifus — APPROVED** [review]

**Comment by jaronoff97** [comment]
@pavolloffay these two tests keep failing, but i do not know why... any ideas?

**Comment by pavolloffay** [comment]
> 2024-10-15T19:33:14.0506287Z go: downloading github.com/kyverno/chainsaw v0.2.11
2024-10-15T19:33:14.3746915Z go: github.com/kyverno/chainsaw@v0.2.11 requires go >= 1.23.0; switching to go1.23.2
2024-10-15T19:33:14.3782275Z go: -d flag is deprecated. -d=true is a no-op 

I will use a lower version

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -62,8 +62,6 @@ jobs:
       with:
         path: bin
         key: ${{ runner.os }}-${{ runner.arch }}-${{ hashFiles('Makefile') }}-${{ steps.setup-go.outputs.go-version }}
-    - name: Install chainsaw
```
Comment (pavolloffay) [inline .github/workflows/e2e.yaml:65]: the way our makefile is written this is not needed, actually it confuses which version is used.

File: Makefile [hunk 3]
```diff
@@ -473,7 +473,7 @@ KUSTOMIZE_VERSION ?= v5.0.3
 CONTROLLER_TOOLS_VERSION ?= v0.16.1
 GOLANGCI_LINT_VERSION ?= v1.57.2
 KIND_VERSION ?= v0.20.0
-CHAINSAW_VERSION ?= v0.2.5
+CHAINSAW_VERSION ?= v0.2.8
```
Comment (pavolloffay) [inline Makefile:476]: It seems like 0.2.9 requires go 1.23 https://github.com/kyverno/chainsaw/releases/tag/v0.2.9 so we can updated only to 0.2.8

**Pull request merged**